javascript/javascript1/week2/flightFullname.js

```diff
@@ -0,0 +1,32 @@
+// function getFullname(firstname, surname){
+//     return firstname + ' ' + surname;
+// }
+
+
+// // Formal fullname
+// function getFullname(firstname, surname, useFormalName) {
+//     return useFormalName ? 'Lord ' + firstname + ' ' + surname : 
+//     firstname + ' ' + surname;
+// }
+
+
+function getFullname(firstname, surname, useFormalName=true, gender) {
+    return firstname === '' || surname === '' ? 'Please provide your firstname and surname.' : 
```

Comment:
Ternary operators can be cool, but it is so easy to get lost in them. 🥴 I would highly recommend to never nest them inside each other and instead use if-elseif-else. 🪆 This is mainly because it is not intuitive what belongs to what. You can take a look [here](https://stackoverflow.com/questions/73603909/right-associativity-and-order-of-execution-of-nested-ternary-operator-in-c) for someone else who tried to decipher nested ternaries. 😄

Also a good opportunity to try some [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)! 🤩